usaspending_api/search/v2/views/spending_by_category_views/spending_by_federal_account.py

```diff
@@ -0,0 +1,67 @@
+import json
+from abc import ABCMeta
+from decimal import Decimal
+from enum import Enum
+from typing import List
+
+from django.db.models import QuerySet
+
+from usaspending_api.common.exceptions import InvalidParameterException
+from usaspending_api.search.v2.views.spending_by_category_views.spending_by_category import (
+    Category,
+    AbstractSpendingByCategoryViewSet,
+)
+
+
+class AccountType(Enum):
+    FEDERAL_ACCOUNT = "federal_account"
+
+
+class AbstractAccountViewSet(AbstractSpendingByCategoryViewSet, metaclass=ABCMeta):
+    """
+    Abstract class used by Federal Accounts spending_by_category endpoint
+    """
+
+    account_type: AccountType
+
+    def build_elasticsearch_result(self, response: dict) -> List[dict]:
+        results = []
+        account_info_buckets = response.get("group_by_agg_key", {}).get("buckets", [])
+        for bucket in account_info_buckets:
+            account_info = json.loads(bucket.get("key"))
+            results.append(
+                {
+                    "amount": int(bucket.get("sum_field", {"value": 0})["value"]) / Decimal("100"),
+                    "id": int(account_info.get("id")) if account_info.get("id") else None,
+                    "code": account_info.get("federal_account_code") or None,
+                    "name": account_info.get("account_title"),
+                }
+            )
+
+        return results
+
+    def query_django(self, base_queryset: QuerySet) -> List[dict]:
+        if "recipient_id" not in self.filters:
+            raise InvalidParameterException("Federal Account category requires recipient_id in search filter")
+        django_filters = {f"federal_account_id__isnull": False}
```

Comment:
Not that important, but these don't need the `f` because they are actually being used as f-strings.